sos/report/plugins/aap_gateway.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2024 Lucas Benedito <lbenedit@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin
+
+
+class AAPGatewayPlugin(Plugin, RedHatPlugin):
+    short_desc = 'AAP Gateway plugin'
+    plugin_name = 'aap_gateway'
+    profiles = ('sysmgmt', 'ansible',)
+
+    packages = ('automation-gateway',
+                'automation-gateway-config')
+    commands = ('aap-gateway-manage',)
+    services = ('automation-gateway',)
+
+    def setup(self):
+        self.add_copy_spec([
+            "/var/log/nginx/access.log*",
+            "/var/log/nginx/error.log*",
+            "/var/log/supervisor",
+            "/etc/ansible-automation-platform",
+            "/etc/supervisord.d/",
+            "/var/log/ansible-automation-platform/gateway/",
+        ])
+
+        self.add_forbidden_path([
+            "/etc/ansible-automation-platform/gateway/SECRET_KEY",
+            "/etc/ansible-automation-platform/gateway/*.key",
+            "/etc/ansible-automation-platform/gateway/*.cert",
+        ])
+
+        self.add_cmd_output([
+            "aap-gateway-manage list_services",
+            "ls -ll /etc/ansible-automation-platform/",
+            "ls -ll /etc/ansible-automation-platform/gateway/",
+            "umask -p",
```

Comment:
We collect it in AAP_Controller already (https://github.com/sosreport/sos/blob/0a876d47196bf9738b4d14509b593583f59c7e3a/sos/report/plugins/aap_controller.py#L63). It might have sense to collect same command output by two plugins (less rationale against that that colecting same files), but still.. does not it make sense to have that collected on one place?

It might be beneficial to have it collected more "generically", but I cant find a proper plugin fir it. Maybe `host`?

@TurboTurtle , @arif-ali , any idea where to collect `umask -p`?

Reply:
I don't see the `umask -p` command, but I do `umask -S`, could this be RH specific? i would lean towards `filesys` maybe?

Reply:
Based on the packages, commands and services, the aap_gateway and aap_controller would not be executed in the same host. Those components only run on different nodes.
With that said, I can remove the umask from both plugins if we think there is a better place for it.

Reply:
`filesys` seems a good plugin for it. +1 for removing `umask -p` from both plugins and adding it there.

Since Ubuntu does not know `-p` flag and that flag just prepends `umask ` word (such that the output string is a valid command to set umask) and that it does not bring any additional value, let add plain `umask` there.